### ISSUE TITLE
[CRITICAL] Simplecast integration is broken 

### ISSUE THREAD

**Issue by jorgelainfiesta** [thefrontside/frontside.io#131]
## Problem 

This repo contains a custom gatsby plugin (`/plugins/gatsby-source-simplecast`) to gather our podcast episodes and display them as a list (https://frontside.com/podcast/) and pages for each podcast episode (ex. https://frontside.com/podcast/open-telemetry-with-austin-parker).

We had used simplecast v1 by recommendation of Mandy, our podcast producer. However, simplecast sent us an email saying that they would deprecate completely v1 and move our show to v2, which has a different UI and API. Therefore, our integration gastby plugin broke down.

**No deployments of the frontside website can happen until this bux is fixed, as it will lead to a build failure**

## Solution 

The solution that comes to mind first is refactoring the plugin to use simplecast v2 API. However, I don't know how v2 API looks like so I can't tell how much effort it'll take, or if it's even a feasible.

**Comment by cowboyd** [comment]
After digging into this, there is some good news, but mostly bad news.

The good news is that I kinda got it working, 

The bad news is that new simplecast API doesn't have author information on the  `/episodes` REST collection like the old API did. That means that instead of doing one query for our entire catalog, we have to do N+1 queries. In our case this is about 150 queries, which is quite a lot, and slows down the time to render the site considerably.

The other bad news is that this number of queries pushes us over the default rate limit threshold and so we can't actually build the site without introducing artificial delays into the fetch. I've spoken with simplecast about increasing our limit.

And finally, there is a third piece of bad news. The slug format has changed between their legacy api and their new api, and not for the better. Descriptive and SEO friendly slugs have been replaced by a hexidecimal number.

For example, the slug for our latest podcast in the legacy api was `open-telemetry-with-austin-parker`. Now, it's `509620d7` which means our previously search engine optimal `/podcast/open-telemetry-with-austin-parker` is now the far less google juice-y `/podcast/509620d7`. But worse, it means that we need to set up redirects from the old url to the new. 

> Note: we could just start generating our own slugs within gatsby, but we'd still need to manually verify each redirect since there'd be no guarantee that their algorithm and ours would arrive at the same result for every podcast.

I think our gatsby architecture needs a re-think. It's not good that we have source data which is now effectively lost to posterity via the sunsetting of the legacy API. We should never dynamically pull resources from the network and use them directly as a gatsby source. Instead, we should use a script to update a file on disk that is _under source control_ and then source pulls from that static file. This has several benefits

1. You can always render the site, even when you're not connected to the network, or when 3rd party services like simplecast go down.
2. Development is faster, because 99% of the time you aren't even working on network related aspects of the site, so why should you have to query simplecast?
3. If we find ourselves in a situation like this one where we can no longer access the 3rd party service, we nevertheless have a record of what the old value was.

In summary, the new gatsby rule should be "Thou shalt check into source control everything that is needed to render the site"

So here's we have to do now:

- [ ] fix the rate limiting problem
- [ ] implement the slug algorithm on our gatsby code since it is no longer implemented on simplecast
- [ ] manually check each url and set up a redirect for it if it's different

What we can do in the future: use a webhook to trigger a workflow_dispatch event that downloads and submits a PR for the podcast content.

**Comment by jbolda** [comment]
Progress Update:

It sounds as if our rate limit was increased for the time being so we can hopefully brute force query for the time being.

I began work on writing the query out to a file, and building the gatsby nodes from a file. I _believe_ that is working, but it appears that the API key on netlify is not working for me? I receive a `503 Service Temporarily Unavailable` whenever I try to query it with the key. Even a single query fails with this error. I am also unable to login with the credentials in 1Password so I think I will need some of your assistance with this @cowboyd.

It looks like we have the slugifying working in the code Charles committed. I used the sitemap on the production version of our website, and some nushell piping to grab a list of all of the podcast URLs. Whatever we end up with, I can set up redirects for the ones that don't conform. It looks like some of the old URLs had the episode number in it and the new URLs do not. I presume we want to remove the episode number from all of the old URLs?

Simplecast doesn't have any webhooks so the next best option to trigger a website feed update / publish is likely through something like zapier, n8n or a cron job somewhere. It would read the XML from simplecast for changes, and hit a workflow dispatch in github actions to PR the change to the json file. Speaking of the rss feed, it seems that simplecast is having us update the URL for our feed. As we request adding it to other services, I think it would be prudent to have a URL on our website that redirects to their RSS feed. This would give us the option to more easily make host adjustments if necessary.
